### DISCUSSION TITLE
Pip install failed

### DISCUSSION THREAD

**Discussion by djcvan** [neuropoly/axondeepseg#711] [Help needed]
Hello, 
I am new to using ADS and new to using the IOS operating system (more experience with windows). I am unfortunately having trouble moving past the pip install -e . portion of the installation. To start, I am using a coworkers mac who has used ADS on this computer and I activated his environment 'ads_venv'. When I tried to open fsleyes it did not let me do this so I attempted to participate in the steps following the environment activation. When I typed the following code it gave me the following issue:

(ads_venv) Admins-iMac:axondeepseg Hans$ pip install -e .
-bash: pip: command not found

I am not sure where to go from here. I decided to create a new personal environment and got the following code.

note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for opencv-python-headless
ERROR: Could not build wheels for opencv-contrib-python, opencv-python-headless, which is required to install pyproject.toml-based projects

failed

CondaEnvException: Pip failed

At this point, I am not sure where to go. I would appreciate any insight that anyone has into this problem. 

Thank you, 
Dax

**Comment by Stoyan-I-A** [comment]
> ERROR: Failed building wheel for opencv-python-headless
> ERROR: Could not build wheels for opencv-contrib-python, opencv-python-headless, which is required to install pyproject.toml-based projects

Those errors come from a package that we seem to no longer use. We'll try to remove it from our requirements and see if it helps.

**Comment by Stoyan-I-A** [comment]
Also, for your information, I am currently working on creating an AxonDeepSeg plugin for napari, which will be Windows compatible. 
It is not ready yet but you can follow the progress here: https://github.com/axondeepseg/napari-plugin. It should be "Ready for testing" in the upcoming weeks.

**Reply by djcvan** [reply]
This is amazing! I am excited for the finished project

**Comment by Stoyan-I-A** [comment]
We've updated the requirements and removed the opencv related ones. 
Could you try to follow the instructions again?
This time, when you create your environment, you could give it a different name (like ads_venv2)
```
conda env create --name ads_venv2 --file environments.yml
conda activate ads_venv2
```

**Reply by mathieuboudreau** [reply]
It's quite odd that @djcvan had a missing `pip` error, as it's a requirement in our env file - https://github.com/axondeepseg/axondeepseg/blob/ab28b4f506311c6ecc05adc69f056077278523da/environment.yml#L27.

Which means, very likely, the env file was not installed correctly or at all. Doing a `conda list` within the conda environment could confirm this.

**Reply by mathieuboudreau** [reply]
@djcvan have you resolved your issue?

**Reply by djcvan** [reply]
Hey thank you for reaching out. I think it must be a problem with the computer that I am using/installation. I was unable to get it to work. I am switching to a different computer and will try again shortly.

**Reply by djcvan** [reply]
@Stoyan-I-A The line of code that you wrote did not seem to solve the problem making me feel that it is an installation problem
